core/src/main/scala/kafka/server/BrokerMetadataCheckpoint.scala

```diff
@@ -21,38 +21,203 @@ import java.io._
 import java.nio.file.{Files, NoSuchFileException}
 import java.util.Properties
 
+import kafka.common.{InconsistentBrokerMetadataException, KafkaException}
+import kafka.server.RawMetaProperties._
 import kafka.utils._
+import org.apache.kafka.common.Uuid
 import org.apache.kafka.common.utils.Utils
 
-case class BrokerMetadata(brokerId: Int,
-                          clusterId: Option[String]) {
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+object RawMetaProperties {
+  val ClusterIdKey = "cluster.id"
+  val BrokerIdKey = "broker.id"
+  val NodeIdKey = "node.id"
```

Comment:
Yeah, that is the plan. Once KIP-500 is enabled, we will deprecate `broker.id` so that `node.id` is the only ID going forward. That said, I'm trying to give us a little wiggle room by defining this config as internal for now. So we still have an opportunity to change this. If you have any thoughts about it, definitely share them.